src/ImageSharp/Formats/Jpeg/GolangPort/OrigJpegDecoderCore.cs

```diff
@@ -361,12 +361,11 @@ public void ParseStream(Stream stream, bool metadataOnly = false)
 
                         break;
                     case OrigJpegConstants.Markers.SOS:
-                        if (metadataOnly)
+                        if (!metadataOnly)
```

Comment:
After "ignoring" SOS, the decoder will start scanning through the contents of the SOS block using tens (hundreds? thousands?) of `FindNextFileMarker()` calls returning in invalid "file markers". I don't think this is the expected behavior.

Can't this behavior push the decoder into a faulted state leading to ignore some further file markers it shouldn't ignore, even in metadata-only state?

Reply:
Accidentally commented this on `OrigJpegDecoderCore` instead of `PdfJsDecoderCore`. 

Actually I only investigetad the latter, and the issue is present here! Haven't checked `OrigJpegDecoderCore`.

Reply:
Does this marker declare it's length? If so we might be able to simply skip the length.

Reply:
I guess [it should](
http://vip.sugovica.hu/Sardi/kepnezo/JPEG%20File%20Layout%20and%20Format.htm).

However, `ProcessSOS` does not depend on it as far as I understand.

Reply:
We can just skip the remaining value same as the other markers.